students/kegan/session07/test_html_render.py

```diff
@@ -0,0 +1,208 @@
+"""
+Kathryn Egan
+
+Suite of tests for components of html_render.py
+"""
+from html_render import *
+
+
+def test_element_init():
+    element = Element()
+    assert element.content == []
+    element = Element('content')
+    assert element.content == ['content']
+    element = Element('')
+    assert element.content == ['']
+
+
+def test_append():
+    element = Element()
+    element.append('content')
+    element.append('more content')
+    assert element.content == ['content', 'more content']
+
+
+# def test_render_no_indent():
+#     html = Html()
+#     html.append(Body('body content'))
+#     html.append(P())
+#     temp = 'html_test.html'
+#     with open(temp, 'w') as f:
+#         html.render(f)
+#     with open(temp, 'r') as f:
+#         assert f.read() == '<html><body>body content</body><p></p></html>'
+
+
+def prepare(elements):
+    """ Generates the HTML output based on given indents and
+    elements to compare rendered HTML to.
+    Args:
+        elements (list of tuples) : indent, element pairs
+    Returns:
+        str : elements as rendered HTML
+    """
+    output = '\n'.join([
+        '    ' * (indent + 1) + element for indent, element in elements])
+    output = '<!DOCTYPE html>\n{}\n'.format(output)
+    return output
+
+
+def check_render(html, expected):
+    """ Checks results of rendered HTML against expected output."""
+    temp = 'html_test.html'
+    with open(temp, 'w') as f:
+        html.render(f)
+    with open(temp, 'r') as f:
+        assert f.read() == expected
+
+
+def test_render():
+    html = Html()
+    body = Body()
+    body.append(P('paragraph 1'))
+    body.append(P('paragraph 2'))
+    html.append(body)
+    elements = [
+        (0, '<html>'),
+        (1, '<body>'),
+        (2, '<p>'),
+        (3, 'paragraph 1'),
+        (2, '</p>'),
+        (2, '<p>'),
+        (3, 'paragraph 2'),
+        (2, '</p>'),
+        (1, '</body>'),
+        (0, '</html>')]
+    expected = prepare(elements)
+    check_render(html, expected)
+
+
+def test_one_line_tag():
+    html = Html()
+    html.append(Title('this is the title'))
+    elements = [
+        (0, '<html>'),
+        (1, '<title>this is the title</title>'),
+        (0, '</html>')]
+    expected = prepare(elements)
+    check_render(html, expected)
+
+
+def test_self_closing_tag():
```

Comment:
ideally, these would be tested by themselves -- does a self closing tag render itself properly? this is almost that, but it's also testing that the html element does the right thing.